src/eterna/folding/EternafoldThreshknot.ts

```diff
@@ -0,0 +1,172 @@
+import * as log from 'loglevel';
+import EmscriptenUtil from 'eterna/emscripten/EmscriptenUtil';
+import SecStruct from 'eterna/rnatypes/SecStruct';
+import Sequence from 'eterna/rnatypes/Sequence';
+/* eslint-disable import/no-duplicates, import/no-unresolved */
+import * as EternafoldLib from './engines/EternafoldLib';
+import {FullFoldResult} from './engines/EternafoldLib';
+import EternaFold from './Eternafold';
+
+export default class EternaFoldThreshknot extends EternaFold {
+    public static readonly NAME: string = 'EternaFoldThreshknot';
+
+    /**
+     * Asynchronously creates a new instance of the Eternafold folder.
+     * @returns {Promise<EternaFold>}
+     * @description AMW TODO cannot annotate type of module/program; both are any.
+     */
+    public static create(): Promise<EternaFoldThreshknot | null> {
+        // eslint-disable-next-line import/no-unresolved, import/no-extraneous-dependencies
+        return import('engines-bin/eternafold')
+            .then((module) => EmscriptenUtil.loadProgram(module))
+            .then((program) => new EternaFoldThreshknot(program))
+            .catch((_err) => null);
+    }
+
+    protected constructor(lib: EternafoldLib) {
+        super(lib);
+        this._lib = lib;
+    }
+
+    public get name(): string {
+        return EternaFoldThreshknot.NAME;
+    }
+
+    public get canPseudoknot(): boolean {
+        return true;
+    }
+
+    /* override */
+    /**
+     * This overrides Eternafold's default foldSequenceImpl implementation with
+     * a post-fold modification. Here we run Eternafold on the sequence as usual,
+     * then post-process the returned base pairs with the Threshknot heuristic
+     * algorithm (see https://arxiv.org/abs/1912.12796).
+     * @param seq RNA sequence to be folded
+     * @param _structStr structStr
+     * @param _temp Environmental temperature
+     * @param _gamma gamma
+     * @param _theta Parameter to adjust filter sensitivity of Threshknot algorithm
+     * @returns SecStruct object representing secondary structure output of engine
+     */
+    protected foldSequenceImpl(
+        seq: Sequence,
+        _structStr: string | null = null,
+        _temp: number = 37,
+        _gamma: number = 6.0,
+        _theta: number = 0.15
```

Comment:
Parameters should only be prefixed with `_` if unused but needs to be retained due to subclasses/parent classes